x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -9,29 +9,47 @@ import { Feature, GeoJsonProperties } from 'geojson';
 import { ESTermSource } from '../sources/es_term_source';
 import { getComputedFieldNamePrefix } from '../styles/vector/style_util';
 import {
-  META_DATA_REQUEST_ID_SUFFIX,
   FORMATTERS_DATA_REQUEST_ID_SUFFIX,
+  META_DATA_REQUEST_ID_SUFFIX,
+  SOURCE_TYPES,
 } from '../../../common/constants';
-import { JoinDescriptor } from '../../../common/descriptor_types';
+import {
+  ESTermSourceDescriptor,
+  JoinDescriptor,
+  TableSourceDescriptor,
+  TermJoinSourceDescriptor,
+} from '../../../common/descriptor_types';
 import { IVectorSource } from '../sources/vector_source';
 import { IField } from '../fields/field';
 import { PropertiesMap } from '../../../common/elasticsearch_util';
+import { ITermJoinSource } from '../sources/term_join_source';
+import { TableSource } from '../sources/table_source';
+import { Adapters } from '../../../../../../src/plugins/inspector/common/adapters';
+
+function createJoinTermSource(
+  descriptor: Partial<TermJoinSourceDescriptor> | undefined,
+  inspectorAdapters: Adapters | undefined
+): ITermJoinSource | undefined {
+  if (!descriptor) {
+    return;
+  }
+
+  if (descriptor.type === SOURCE_TYPES.ES_TERM_SOURCE) {
```

Comment:
Looks like you will still need to check for `joinDescriptor.right &&	 'indexPatternId' in joinDescriptor.right && 'term' in joinDescriptor.right`. I had missed that `_rightSource` is optional and an instance can be created without it. Without this check, the UI crashes if you try to add a term join.